tests/e2e/Services/Functions/FunctionsCustomClientTest.php

```diff
@@ -159,6 +159,11 @@ public function testCreateCustomExecution(): array
         $this->assertNotEmpty($output['APPWRITE_FUNCTION_JWT']);
         $this->assertEquals($this->getProject()['$id'], $output['APPWRITE_FUNCTION_PROJECT_ID']);
 
+        $executionId = $execution['body']['$id'] ?? '';
+        $this->assertNotEmpty($output['APPWRITE_FUNCTION_EXECUTION_ID']);
+        $this->assertEquals($executionId, $output['APPWRITE_FUNCTION_EXECUTION_ID']);
```

Comment:
I would do 1 more assertion, in async flow. async execution, or event trigger, or schedule, anywhere.

There we should assert execution ID exists, but client IP is empty